src/sentry/digests/notifications.py

```diff
@@ -19,26 +19,38 @@
 Notification = namedtuple("Notification", "event rules")
 
 
-def split_key(key: str) -> tuple[Project, ActionTargetType, str | None]:
-    key_parts = key.split(":", 4)
+def split_key(
+    key: str,
+) -> tuple[Project, ActionTargetType, str | None, FallthroughChoiceType | None]:
+    key_parts = key.split(":", 5)
     project_id = key_parts[2]
     # XXX: We transitioned to new style keys (len == 5) a while ago on
     # sentry.io. But self-hosted users might transition at any time, so we need
     # to keep this transition code around for a while, maybe indefinitely.
-    if len(key_parts) == 5:
+    if len(key_parts) == 6:
         target_type = ActionTargetType(key_parts[3])
         target_identifier = key_parts[4] if key_parts[4] else None
+        fallthrough_choice = FallthroughChoiceType(key_parts[5])
+    elif len(key_parts) == 5:
+        target_type = ActionTargetType(key_parts[3])
+        target_identifier = key_parts[4] if key_parts[4] else None
+        fallthrough_choice = None
     else:
         target_type = ActionTargetType.ISSUE_OWNERS
         target_identifier = None
-    return Project.objects.get(pk=project_id), target_type, target_identifier
+        fallthrough_choice = None
+    return Project.objects.get(pk=project_id), target_type, target_identifier, fallthrough_choice
```

Comment:
What I'd recommend here is splitting out the `split_key` changes into a separate pr and merge/deploy them first.

The problem you're going to run into is that `unsplit_key` will start sending new keys through, but during the deploy they might be processed by tasks that don't have the new `split_key` code. So they'll fall through to the else block and fail to extract the `target_identifier`.